### PR TITLE
Implements error handling for imap errors #11

### LINKED ISSUE #11: user_external should log error message if authentication over imap does not work
### Affected apps
user_external

### Expected behaviour
If user_external can not login over imap show a message what the error was!

### Actual behaviour
It just shows a general purpose error message, that login was not successful.

It took me over an hour to find out, that the user_external app was properly enabled and even executed, just that the server certificate was wrong.
I managed to find that out by adding only adding a single line to imap.php

OCP\Util::writeLog('user_external', implode('|', imap_errors()), OCP\Util::ERROR);

### Steps to reproduce

1.    activate user_external app
2.    change config.php to activate login over imap (but have an error in your configuration)
3.    try to login

### PR DESCRIPTION
**Pull request by anojht**
Prints IMAP errors and alerts to the nextcloud log. See #11
closes #11

### CODE REVIEW

**Comment by violoncelloCH** [comment]
thank you for your contribution!

DCO is failing because the commit is not signed-off
@anojht could you please sign your commit? - see https://github.com/nextcloud/server/blob/master/.github/CONTRIBUTING.md#sign-your-work
can be done with `git commit --amend -s` for the last commit and then `git push -f` to push the changed commit back to github...

**Comment by anojht** [comment]
@violoncelloCH the commits have been signed

**Comment by violoncelloCH** [comment]
thank you @anojht 
please review @nextcloud/user_external

**Comment by violoncelloCH** [comment]
good point @cweiske

**Comment by anojht** [comment]
@violoncelloCH I have stored the errors and checked for empty list, before printing.

File: lib/imap.php [hunk 1]
```diff
@@ -67,8 +67,20 @@ public function checkPassword($uid, $password) {
  		}
 
 		$mbox = @imap_open($this->mailbox, $username, $password, OP_HALFOPEN, 1);
-		imap_errors();
-		imap_alerts();
+		$imapErrors = imap_errors();
+		$imapAlerts = imap_alerts();
+		if (!empty($imapErrors) {
```
Comment (violoncelloCH) [inline lib/imap.php]: there is a closing ')' missing

Comment (anojht) [reply lib/imap.php]: Whoops, just fixed this. Nice to know it works as intended 😄

**Comment by violoncelloCH** [comment]
@MorrisJobke I just thought another review from one of you could be good, because I'm not that experienced yet... but I tested it once more and think it's fine, so merging it...

**Pull request merged**